public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx

```diff
@@ -1,92 +1,276 @@
+                      type="value"
+                    />
+                  </Field>
+
+                  <RemoveButton className={styles.deleteLabelButton} index={index} remove={remove} />
+                </div>
+              </div>
+            );
+          })}
+          <AddButton className={styles.addLabelButton} append={append} />
+        </>
+      )}
+    </>
+  );
+};
+
+const LabelsWithoutSuggestions: FC = () => {
+  const styles = useStyles2(getStyles);
+  const {
+    register,
+    control,
+    watch,
+    formState: { errors },
+  } = useFormContext<RuleFormValues>();
+
+  const labels = watch('labels');
+  const { fields, remove, append } = useFieldArray({ control, name: 'labels' });
+
+  return (
+    <>
+      {fields.map((field, index) => {
+        return (
+          <div key={field.id}>
+            <div className={cx(styles.flexRow, styles.centerAlignRow)} data-testid="alertlabel-input-wrapper">
+              <Field
+                className={styles.labelInput}
+                invalid={!!errors.labels?.[index]?.key?.message}
+                error={errors.labels?.[index]?.key?.message}
+              >
+                <Input
+                  {...register(`labels.${index}.key`, {
+                    required: { value: !!labels[index]?.value, message: 'Required.' },
+                  })}
+                  placeholder="key"
+                  data-testid={`label-key-${index}`}
+                  defaultValue={field.key}
+                />
+              </Field>
+              <InlineLabel className={styles.equalSign}>=</InlineLabel>
+              <Field
+                className={styles.labelInput}
+                invalid={!!errors.labels?.[index]?.value?.message}
+                error={errors.labels?.[index]?.value?.message}
+              >
+                <Input
+                  {...register(`labels.${index}.value`, {
+                    required: { value: !!labels[index]?.key, message: 'Required.' },
+                  })}
+                  placeholder="value"
+                  data-testid={`label-value-${index}`}
+                  defaultValue={field.value}
+                />
+              </Field>
+              <RemoveButton className={styles.deleteLabelButton} index={index} remove={remove} />
+            </div>
+          </div>
+        );
+      })}
+      <AddButton className={styles.addLabelButton} append={append} />
+    </>
+  );
+};
+
+const LabelsField: FC<Props> = ({ className, dataSourceName }) => {
+  const styles = useStyles2(getStyles);
 
   return (
     <div className={cx(className, styles.wrapper)}>
-      <Label>Custom Labels</Label>
+      <Label>
+        <Stack gap={0.5}>
+          <span>Custom Labels</span>
+          <Tooltip
+            content={
+              <div>
+                Labels shown as suggestions in the dropdowns come from previously entered labels in other alert rules.
+                They do not represent the full set of labels for the alert.
```

Comment:
@brendamuir do you agree with this text? Or do you have any suggestions to change it? Basically this means we're not showing autogenerated labels but only the ones that were manually entered previously.

Reply:
The dropdown only displays labels that you have previously used for alerts. 
Select a label from the dropdown or type in a new one.